js/admin.js

```diff
@@ -166,27 +170,32 @@ jQuery( document ).ready(
 		 *
 		 * @param {Object} languageSelected - jQuery object of the selected element in the language list dropdown.
 		 */
-		function fillLanguageFields( languageSelected ) {
+		function fillLanguageFields( language ) {
 			// All field values is obtained by splitting the selected element value in the language list dropdown.
-			var fieldValues = languageSelected.val().split( ':' );
+			var fieldValues = language.values;
 			// Language name field is obtained by splitting the selected element text in the language list dropdown.
-			var languageName = languageSelected.text().split( ' - ' );
+			var languageName = language.name;
 			$( '#lang_slug' ).val( fieldValues[0] );
 			$( '#lang_locale' ).val( fieldValues[1] );
 			$( 'input[name="rtl"]' ).val( [fieldValues[2]] );
 			$( '#lang_name' ).val( languageName[0] );
+		}
 
-			// Refresh the flag field only if it's present.
-			if ( flagListExist ) {
-				$( '#flag_list').val( fieldValues[3] );
-				// Refresh the jQuery UI selectmenu flag list widget by triggerring its change event.
-				selectmenuFlagList._trigger( 'change' );
-			}
-		};
+		function parseSelectedLanguage( event ) {
+			var selectedElement = $('option:selected', event.target);
+			return {
+				values: selectedElement.val().split(':'),
+				name: selectedElement.text().split(' - ')
+			};
+		}
 
 		// Callback when selectmenu widget change event is triggered.
 		var changeCallback = function( event, ui ) {
-			fillLanguageFields( $( 'option:selected', event.target ) );
+			var language = parseSelectedLanguage( event );
+
+			fillLanguageFields( language );
+
+			$( event.target ).trigger( 'notifyChange', language );
```

Comment:
Same comment as in the first one, we could push only the language flag value to the `notifyChange` event to modify the flag field.